python/ray/util/client/worker.py

```diff
@@ -431,6 +438,62 @@ def ping_server(self) -> bool:
     def is_connected(self) -> bool:
         return self._conn_state == grpc.ChannelConnectivity.READY
 
+    def _server_init(self, job_config: JobConfig):
+        """Initialize the server
+
+        There are two steps communication between client and server.
+        First request is `Init`, which will call `ray.init` in
+        server side. The server will return a status which might
+        trigger the second request:
+          Status.OK: it means the serer is ready to accept request.
+              No future work is needed.
+          Status.INCOMPATIBLE_RUNTIME_ENV: it means the server is
+              having different runtime env as the client, the client
+              won't be able to run in this server.
+          Status.MISSING_URIS: it means server missed some uris and
+              client need to prepare them. After the preparation,
+              it need to send another request `PrepRuntimeEnvRequest`
+              to notify the server to fetch it.
+
+        Args:
+            job_config (JobConfig): Job config to pass to the server
+        """
+        import ray._private.runtime_env as runtime_env
+        import tempfile
+        with tempfile.TemporaryDirectory() as tmp_dir:
+            if runtime_env.PKG_DIR is None:
+                runtime_env.PKG_DIR = tmp_dir
+            # Generate the uri for runtime env
+            if job_config is None:
+                job_config = JobConfig()
+            runtime_env.rewrite_working_dir_uri(job_config)
```

Comment:
Instead of adding a special protocol to check if the package exists, can you instead use the runtime_env methods like package_exists() directly? Internal KV works over Ray client already, so those methods should all work.